src/main/webapp/app/routes/routes.tsx

```diff
@@ -166,13 +167,19 @@ const AppRouts = (props: {
             isUserAuthenticated={props.authenticationStore.isUserAuthenticated}
             appStore={props.appStore}
             path={PAGE_ROUTE.GENE}
+            canonicalProps={{
+              __canonicalTypeName: 'location',
```

Comment:
to denote it's only need to help typescript with type hinting.
It's a convention I stole from a graphql library I used for CTDataHub.

Reply:
Basically I'm trying to know what object I'm looking at. If I use the "typeof" or "instanceof" keywords it will only tell me that it's an object. I could make a class with instanceof, but that seems like overkill so what I'm doing instead is telling typescript that there can only be a handful of hardcoded strings which are valid in the "canonicalTypeName" field. Since each string is unique to each object type, performing an equality check tells typescript exactly which type you're looking at.

[Here's an example](https://www.typescriptlang.org/play/?#code/FAFwngDgpgBACgQxASygOxDAvDA3sGQmAfWPGgDkEBbKALhgCIIlUNGAaAogE2QQDmaAPYBnZKIaiQAJ2RoBwAL7BQkWABFhAYxDCZ2PN0KlyUKrQaMeOvTM7GYLEFHkhJ8VuhABtALrKqmbwUDKiwmiGiCjeMAA+MFq6+tzA2hHSTlDCEAA29CFhEf6GPo64JGTqFgXMXuwcMHyCIuIejAAWUAi5IB1gjDBKXER4lWY1VjbJ9o3OrhgeZaOj+Cvr49U0tc5sIA4bK81CYhJW4toA1gcbKqN+KgHA0Dn5AHQAZvoAogjaHQAKaBFSJYAB8RlGyA+MCBoXCaDepi2tGwWBwdRi7AAlJCNuk0OF3rlhAI4SC3sdWhJsQBuRxKGBQXKiWBrfEZYTE0nkhFveZuUR0hnKbGqF55KCfH5-QHAhHYCHswjQ2HyiJIqqUbZojG7byMXHKlYEolSklk9WIqmnIX00YAegdMDMom0cggmBEAHcYJcfaIXR0JFkQTAQwgmjphHYYAg0DwYOE40x9ewXepw4HkNQIGJxAAjfIMpks2Cq3ka5Ha1HovX1fZGxyjU1c808q38pALdzC0YqJRi57ZSXSmS-f6V0FKxwVzvV8w6uuphuGvHrVvcy3wjU2tp9oiM5ms8MwqeaiZL9FMaZ2NfGlucrfngWLA+EI9l9crJ2hhURmA0CgAA3UIM2gGACygbQEAAVxPZAQAAcmzXN82QItYD0GAZG6f4gxDdIeCgZsiE3dttxBd8hlFIA)

Reply:
Maybe I could use the ["in" operation instead](https://www.turing.com/kb/check-type-of-objects-variables-in-typescript)